rules/framework.bzl

```diff
@@ -64,14 +64,18 @@ def apple_framework(name, apple_library = apple_library, **kwargs):
         "//conditions:default": None,
     }))
 
+    # We need to put back bundle_id in order to set that attr for 'precompiled_apple_resource_bundle' targets in 'apple_library'
```

Comment:
It wouldn't be the same, it would have the same fmw identifier as prefix + the bundle_name for the specific resource, grep this line in my PR:
```py
 bundle_id = "{}.{}".format(bundle_id, bundle_name)
```

Tbh I wasn't super sure how to setup this, what I implemented here gives us a unique value unless I'm missing something. But regarding will also take a look at what cocoapods does and compare, thx for calling it out!

Reply:
@jerrymarino see the [latest commit](https://github.com/bazel-ios/rules_ios/pull/475/commits/e902672e910380d97178c288f479610fab100e56):

[This value was already being set before and I didn't realize it](https://github.com/bazel-ios/rules_ios/blob/76a1551baf12f1784873be4e31372a2a29c5405b/rules/precompiled_apple_resource_bundle.bzl#L106), glad you caught this otherwise I'd be changing the value for the bundle identifier here :)

That said, the only thing we need is to move this assignment up to the rule declaration so aspects can collect it. Added to a utils file since it's used in two places, will make it easier to make future changes to this logic

Reply:
Also, no need to propagate the fmw `bundle_id`, that was just my first attempt to compose an identifier. I'll keep the fmw test though (without setting the bundle identifier) because we will eventually run the XCHammer targets as part of the tests here and we want to cover that test case

Reply:
I think this is the right path to understand why we ended up having to collect this directly from an attribute, will track this separately and tackle that in a follow up

Also don't have the context of past decisions on that rule but can def investigate